### PR TITLE
Let's play spot the bug

### PR DESCRIPTION
**Pull request by nshalman**
There's a bug in the code. `port_associate` can't find the file whose path I'm pretty sure I've put into the FileObj struct.
I can read the path back out to golang again, and looking at the bytes it sure looks like the right string of characters.

What am I missing?

```shell
$ go test -v syscall_illumos_test.go
=== RUN   TestLifreqSetName
--- PASS: TestLifreqSetName (0.00s)
=== RUN   TestCreateFileObj
--- PASS: TestCreateFileObj (0.00s)
=== RUN   TestBasicEventPort
    syscall_illumos_test.go:65: PortAssociateFileObj failed: no such file or directory
--- FAIL: TestBasicEventPort (0.00s)
FAIL
FAIL    command-line-arguments  0.005s
FAIL
```

### CODE REVIEW

File: unix/syscall_illumos.go [hunk 1]
```diff
@@ -176,3 +178,55 @@ func (s *Strioctl) SetInt(i int) {
...
+func PortCreate() (int, error) {
+	return port_create()
+}
+
+//sys	port_associate(port int, source int, object uintptr, events int, user unsafe.Pointer) (n int, err error)
+
+//TODO
+//func PortAssociateFd(port int, fd int, events int, user unsafe.Pointer) (n int, err error)
+
+func PortAssociateFileObj(port int, f *FileObj, events int, user unsafe.Pointer) (int, error) {
```
Comment (Toasterson) [inline unix/syscall_illumos.go:195]: Blind guess double pointer?

Comment (Toasterson) [reply unix/syscall_illumos.go:195]: I can't for the life of me remember how pointer conversion from pointer to uintptr works. Maybe unsafe.Pointer is the wrong converter.

Comment (nshalman) [reply unix/syscall_illumos.go:195]: How would I fix it?

**Comment by nshalman** [comment]
Partial solution:
```diff
diff --git a/unix/syscall_illumos.go b/unix/syscall_illumos.go
index fd75706..2c556c9 100644
--- a/unix/syscall_illumos.go
+++ b/unix/syscall_illumos.go
@@ -211,7 +211,7 @@ func CreateFileObj(name string, stat os.FileInfo) (*FileObj, error) {
        if err != nil {
                return nil, err
        }
-       fobj.Name = (*int8)(unsafe.Pointer(&bs))
+       fobj.Name = (*int8)(unsafe.Pointer(&bs[0]))
        fobj.Atim.Sec = stat.Sys().(*syscall.Stat_t).Atim.Sec
        fobj.Atim.Nsec = stat.Sys().(*syscall.Stat_t).Atim.Nsec
        fobj.Mtim.Sec = stat.Sys().(*syscall.Stat_t).Mtim.Sec
```

**Comment by nshalman** [comment]
Thanks again @Toasterson!

**Pull request closed without merging**